### PR TITLE
UCP: Fix RNDV_SEND_NBR_THRESH setting

### PR DESCRIPTION
**Pull request by brminich**
RNDV was not used for send_nbr, because it was adjusted to SIZE_MAX.
In case of HW TM it was adjusted to maximal segment size.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6463/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6465/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6468/ for details (Mellanox internal link).

**Comment by brminich** [comment]
@alex-mikheev, plz take a look

File: src/ucp/core/ucp_context.c [hunk 1]
```diff
@@ -98,7 +98,8 @@ static ucs_config_field_t ucp_config_table[] = {
    ucs_offsetof(ucp_config_t, ctx.rndv_thresh), UCS_CONFIG_TYPE_MEMUNITS},
 
   {"RNDV_SEND_NBR_THRESH", "256k",
-   "Threshold for switching from eager to rendezvous protocol in ucp_tag_send_nbr()",
+   "Threshold for switching from eager to rendezvous protocol in ucp_tag_send_nbr().\n"
+   "Relevant when UCX_RNDV_THRESH is set to auto only.",
```
Comment (yosefe) [inline src/ucp/core/ucp_context.c]: Relevant only if UCX_RNDV_THRESH is set to \"auto\"

File: src/ucp/core/ucp_ep.c [hunk 2]
```diff
@@ -838,22 +825,26 @@ static void ucp_ep_config_set_rndv_thresh(ucp_worker_t *worker,
...
+                                                   iface_attr->cap.get.min_zcopy,
+                                                   max_rndv_thresh);
+
+    config->tag.rndv_send_nbr.rma_thresh = ucp_ep_thresh(rndv_nbr_thresh,
+                                                         iface_attr->cap.get.min_zcopy,
+                                                         max_rndv_thresh);
 
-    ucp_ep_config_set_rndv_send_nbr_thresh(context, config, adjust_min_val,
-                                           iface_attr->cap.get.max_zcopy);
+    ucs_trace("Rndv threshold is %zu (send_nbr: %zu)",
```
Comment (yosefe) [inline src/ucp/core/ucp_ep.c]: Rndv -> rndv

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6480/ for details (Mellanox internal link).

**Pull request merged**